js/exporter/image_creator.js

```diff
@@ -419,31 +419,44 @@ function drawElement(element, context, parentOptions, shared) {
 
     if(!isText) {
         applyFilter(context, options, shared);
-        fillElement(context, options, shared);
+        if(!isImage) {
+            promise = fillElement(context, options, shared);
+        }
         strokeElement(context, options);
     }
 
-    applyGradient(context, options, shared, element);
+    applyGradient(context, options, shared, element, 'linear');
+    applyGradient(context, options, shared, element, 'radial');
 
     context.restore();
 
     return promise;
 }
 
-function applyGradient(context, options, { gradients }, element) {
-    if(gradients.length === 0) {
+function applyGradient(context, options, { linearGradients, radialGradients }, element, type) {
+    const gradients = type === 'linear' ? linearGradients : radialGradients;
+    if(Object.keys(gradients).length === 0) {
         return;
     }
     const id = parseUrl(options.fill);
     if(id && gradients[id]) {
         const box = element.getBBox();
-        const gradient = context.createLinearGradient(box.x, 0, box.x + box.width, 0);
+        const horizontalCenter = box.x + box.width / 2;
+        const verticalCenter = box.y + box.height / 2;
+        const gradient = type === 'linear' ?
+            context.createLinearGradient(box.x, 0, box.x + box.width, 0) :
```

Comment:
with the help of the third parameter, do we pass the final coordinate or is it the width?

(I saw that this is an old code)

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createLinearGradient